File: src/main/java/me/mikael/shopping/api/domain/Shop.java
package me.mikael.shopping.api.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import me.mikael.shopping.api.distance.LocationAware;

import javax.persistence.*;

/**
 *
 */
@Entity
@Data
public class Shop implements LocationAware{

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Version
    @JsonIgnore
    private long version;

    private String name;

    @OneToOne(cascade = CascadeType.ALL)
    private ShopAddress address;

    @OneToOne(cascade = CascadeType.ALL)
    private Location location;
}

File: shopping-web-a2/src/app/shop.ts
import {Location} from './location';
import {ShopAddress} from "./shop-address";

export class Shop{
  id: number;
  name: string;
  location: Location;
  address: ShopAddress;

  constructor(){
    this.location = new Location();
    this.address = new ShopAddress();
  }
}
